### PR TITLE
always wait for route to return 200 before proceeding

### PR DESCRIPTION
**Pull request by jim-minter**
Fixes https://github.com/openshift/openshift-azure/issues/718

### CODE REVIEW

File: test/e2e/end_user_test.go [hunk 1]
```diff
@@ -79,11 +79,21 @@ var _ = Describe("Openshift on Azure end user e2e tests [EndUser]", func() {
...
 		}
 		By(fmt.Sprintf("hitting the route and checking the contents (%v)", time.Now()))
-		resp, err := httpc.Get(url)
-		Expect(err).NotTo(HaveOccurred())
-		defer resp.Body.Close()
-		Expect(resp.StatusCode).Should(Equal(http.StatusOK))
-		contents, err := ioutil.ReadAll(resp.Body)
+
+		var contents []byte
+		err = wait.PollImmediate(time.Second, 10*time.Second, func() (bool, error) {
```
Comment (thekad) [inline test/e2e/end_user_test.go]: There is a util function `ForHTTPStatusOk()` which will help here (pretty sure @asalkeld wrote it previously)

File: test/e2e/end_user_test.go [hunk 2]
```diff
@@ -142,6 +152,19 @@ var _ = Describe("Openshift on Azure end user e2e tests [EndUser]", func() {
 				Timeout: 10 * time.Second,
 			}
 
+			err := wait.PollImmediate(time.Second, 10*time.Second, func() (bool, error) {
```
Comment (thekad) [inline test/e2e/end_user_test.go]: ditto

File: test/e2e/end_user_test.go (hunk 1 above)
Comment (kargakis) [reply test/e2e/end_user_test.go]: https://github.com/openshift/openshift-azure/blob/2df74f675fd99c4707d02b96c6f2c2c2dc209ed0/pkg/util/wait/wait.go#L47

**Comment by jim-minter** [comment]
thanks @thekad - updated :)

**Pull request merged**